src/main/java/net/mcreator/workspace/elements/VariableElement.java

```diff
@@ -38,10 +43,32 @@
 	}
 
 	public VariableElementType getType() {
-		return type;
+		return getVariableFromType(type);
+	}
+
+	public static VariableElementType getVariableFromType(String type) {
+		for (VariableElementType varType : variables.keySet()) {
+			if(varType.getBlocklyVariableType().equalsIgnoreCase(type) || varType.getName().equalsIgnoreCase(type)
+					|| varType.getBlockName().equalsIgnoreCase(type)) {
+				return varType;
+			}
+		}
+		return null;
 	}
 
-	public void setType(VariableElementType type) {
+	public static Set<VariableElementType> getVariables() {
+		return variables.keySet();
+	}
+
+	public static void addVariable(VariableElementType var) {
+		variables.put(var, var.getName().toUpperCase());
+	}
+
+	public static Set<VariableElementType> getAllTypes() {
+		return variables.keySet();
+	}
+
+	public void setType(String type) {

```

Comment:
I think this method should take VariableElementType and then convert to string inside here.